app/services/invite_creator.rb

```diff
@@ -0,0 +1,25 @@
+class InviteCreator
+  def initialize(email)
+    @email = email
+  end
+
+  def perform!
+    Invite.create!(
+        email: @email,
+        code: generate_code
+    ).tap do |invite|
+      Notifier.beta_invite(invite).deliver_later
+    end
+  end
+
+  private
+  def generate_code
+    candidate = SecureRandom.hex(16)
+    if Invite.exists?(code: candidate)
+      # Collision? Try again.
+      generate_code
```

Comment:
16^16 = 18,446,744,073,709,551,616 combinations. If it recurses even once in a thousand years it would be an epic fluke 😝